### ISSUE TITLE
[Bug]: mo crash by error: error: out of memory when run tpcc 10w 10 terminals test for about 8 hours

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#8984]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):701b6f16d353eb3cac683f2eb73728d3f8bd8b15
- Hardware parameters:16C 64G
- OS type:
- Others:
```

### Actual Behavior

run tpcc 10w 10 terminals test for about 8 hours on standlone mode for about 8 hours, the mo crashed.

9-f9774fbc-d930-11ed-8db4-525400777c3a-702-0:0-0>]"}
2023/04/12 20:59:13.063601 +0800 INFO frontend/util.go:415 time of Exec.Run : 283.045851ms connectionId 1066|127.0.0.1:52030|{account sys:dump:moadmin -- 0:1:0}|6795cf61-d8f7-11ed-8da4-525400777c3a
2023/04/12 20:59:13.063673 +0800 INFO frontend/util.go:415 time of SendResponse 120ns connectionId 1066|127.0.0.1:52030|{account sys:dump:moadmin -- 0:1:0}|6795cf61-d8f7-11ed-8da4-525400777c3a
2023/04/12 20:59:13.063760 +0800 INFO frontend/util.go:400 query trace status {"connection_id": 1066, "statement": "execute __mo_stmt_id_8 using @__mo_stmt_var_0,@__mo_stmt_var_1,@__mo_stmt_var_2,@__mo_stmt_var_3,@__mo_stmt_var_4", "status": "success", "span": {"trace_id": "d2914f14-d931-11ed-8db4-525400777c3a", "kind": "statement"}, "session_info": "connectionId 1066|127.0.0.1:52030|{account sys:dump:moadmin -- 0:1:0}|6795cf61-d8f7-11ed-8da4-525400777c3a"}
2023/04/12 20:59:13.063879 +0800 INFO frontend/util.go:400 query trace {"connection_id": 1066, "query": "execute __mo_stmt_id_8 using @__mo_stmt_var_0,@__mo_stmt_var_1,@__mo_stmt_var_2,@__mo_stmt_var_3,@__mo_stmt_var_4", "vars": "13 , 9 , 0 , 6 , 50131", "session_info": "connectionId 1066|127.0.0.1:52030|{account sys:dump:moadmin -- 0:1:0}|6795cf61-d8f7-11ed-8da4-525400777c3a"}
2023/04/12 20:59:13.064256 +0800 INFO frontend/session.go:369 session uuid: 6795cf61-d8f7-11ed-8da4-525400777c3a -> background session uuid: d2bdf447-d931-11ed-8db4-525400777c3a
2023/04/12 20:59:13.064647 +0800 INFO blockio/writer.go:147 [WriteEnd] {"operation": "name: 85f0f3f9-d931-11ed-8db4-525400777c3a_66, block count: 2, size: 266671"}
2023/04/12 20:59:13.065652 +0800 ERROR mpool/mpool.go:511 error: error: out of memory
github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Alloc
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/common/mpool/mpool.go:511
github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Realloc
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/common/mpool/mpool.go:592
github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Grow
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/common/mpool/mpool.go:658
github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Grow2
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/common/mpool/mpool.go:671
github.com/matrixorigin/matrixone/pkg/container/types.BuildVarlena
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/container/types/bytes.go:73
github.com/matrixorigin/matrixone/pkg/container/vector.appendOneBytes
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/container/vector/vector.go:1320
github.com/matrixorigin/matrixone/pkg/container/vector.AppendAny
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/container/vector/vector.go:1214
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/containers.(*vector[...]).Append
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/vm/engine/tae/containers/vector.go:93
"a.txt" 200000L, 30586785C
[tpcc_out_mem.tar.gz](https://github.com/matrixorigin/matrixone/files/11218499/tpcc_out_mem.tar.gz)

### Steps to Reproduce

```Markdown
1. run a mo cluster on eks with (2*cn, 1*dn, 3*log)
2. git clone https://github.com/matrixorigin/mo-tpcc.git
3. modify props.mo, set warhouse to 10 , terminals to 10, runMins to 5000
4. generate tpcc 10w data, ./runLoader.sh props.mo warehourse 10 filelocaton XXXX
5. create tpcc schema using file sql/tableCreates.sql
6. load data to the mo 
7. run tpcc test, ./runBenchmark.sh props.mo
```

**Comment by ouyuanning** [comment]
bug happend in pull mode

**Comment by nnsgmsone** [comment]
same as #7891

**Comment by nnsgmsone** [comment]
same as #7891

**Comment by nnsgmsone** [comment]
This problem may require a storage person to help look at it, because dn and cn are not using the same mpool.

**Comment by nnsgmsone** [comment]
may be a memory leak of memtabl, let me check.

**Comment by nnsgmsone** [comment]
This issue will be significantly improved after pr 9069 is filed.

**Comment by nnsgmsone** [comment]
This issue will be significantly improved after pr 9069 is filed.

**Comment by aressu1985** [comment]
testing

**Comment by aressu1985** [comment]
on test

**Comment by aressu1985** [comment]
wait for test resoure released

**Comment by aressu1985** [comment]
fixed: 2290b1d3082914bb6b425b4d4dfdf2b50b7029c0
